test/e2e/flowaggregator_test.go

```diff
@@ -192,6 +197,38 @@ func testHelper(t *testing.T, data *TestData, podAIPs, podBIPs, podCIPs, podDIPs
 
 	// OVS userspace implementation of conntrack doesn't maintain packet or byte counter statistics, so we ignore the bandwidth test in Kind cluster.
 	checkBandwidth := testOptions.providerName != "kind"
+
+	// RestartFlowAggregator tests the case, where Flow Aggregator stops, Flow Exporter should be able
+	// to detect and stop sending records until Flow Aggregator restarts.
+	t.Run("RestartFlowAggregator", func(t *testing.T) {
+		// In Kind cluster, there are issues with DNS name resolution on worker nodes.
+		// Flow Exporter needs to be restarted every time Flow Aggregator get restarts since it uses
+		// cluster IP as collector address instead of DNS name.
+		if testOptions.providerName == "kind" {
+			t.Skip("Skipping test because Flow Exporter needs to be restarted anyways in Kind cluster.")
+		}
+		err := testData.deleteFlowAggregator()
+		if err != nil {
+			t.Errorf("Error when deleting Flow Aggregator %v", err)
+		}
+		antreaPodName, err := data.getAntreaPodOnNode(workerNodeName(1))
+		if err != nil {
+			t.Errorf("Error when getting Antrea Pod name %v", err)
+		}
+		now := time.Now().Format(time.RFC3339)
+		errMsg := "Error when initializing flow exporter"
```

Comment:
could we just check if flow records are properly received or not by sending simple traffic after the restart of flow aggregator?

Reply:
In that case the test will take a long time to execute because we need to wait for the flow records to be exported from Flow Exporter, aggregated by Flow Aggregator and then show in the collector log. After re-deployment of flow aggregator, if following tests get passed, we can tell that Flow Exporter works well. That's why I put this test in front of all other tests. Here we actually check whether Flow Exporter can detect the disconnection.

Reply:
I was just thinking about the scenario where Flow Aggregator and Flow Exporter has some state before it gets restarted. Maybe move this to the end.. have a simple test with curl like Pod-To-External flows that take less time or you could keep the current test, where it looks for that particular error log.

Reply:
Makes sense to me. Move ToExternal test to after restart test to avoid duplicate tests on same item.